.github/workflows/push.yml

```diff
@@ -0,0 +1,170 @@
+name: Push
+
+env:
+  IMAGE_NAME: ghcr.io/${{ github.repository_owner }}/cloudigrade
+  NGINX_IMAGE_NAME: ghcr.io/${{ github.repository_owner }}/cloudinginx
+
+on: [push]
+
+jobs:
+  test-py38:
+    name: py38
+    runs-on: ubuntu-latest
+    container:
+      image: docker.pkg.github.com/cloudigrade/build-container/build-container:latest
+      credentials:
+         username: ${{ github.actor }}
+         password: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: pip install tox codecov poetry
+      - run: tox -e py38
+      - run: codecov
+
+  test-flake8:
+    name: flake8
+    runs-on: ubuntu-latest
+    container:
+      image: docker.pkg.github.com/cloudigrade/build-container/build-container:latest
+      credentials:
+         username: ${{ github.actor }}
+         password: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: pip install tox codecov poetry
+      - run: tox -e flake8
+
+  test-vulnerability:
+    name: Vulnerability
+    runs-on: ubuntu-latest
+    container:
+      image: docker.pkg.github.com/cloudigrade/build-container/build-container:latest
+      credentials:
+         username: ${{ github.actor }}
+         password: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: pip install poetry safety
+      - run: poetry install
+      - run: safety check
+
+  test-docker:
+    name: Docker Build
+    runs-on: ubuntu-latest
+    container:
+      image: docker:latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: docker pull $IMAGE_NAME:latest || true
+      - run: docker build --cache-from $IMAGE_NAME:latest .
+
+  test-s2i:
+    name: s2i Build
+    runs-on: ubuntu-latest
+    container:
+      image: docker:latest
+    env:
+      CONTEXT_DIR: deployment/nginx/
+      NGINX_S2I_IMAGE: registry.redhat.io/rhel8/nginx-116:latest
+      S2I_VERSION: v1.3.1
+      S2I_RELEASE: source-to-image-v1.3.1-a5a77147-linux-amd64
+    steps:
+      - uses: actions/checkout@v2
+      - run: apk --no-cache add git
+      - run: mkdir -p /opt && cd /opt
+      - run: wget https://github.com/openshift/source-to-image/releases/download/${S2I_VERSION}/${S2I_RELEASE}.tar.gz -O /opt/s2i.tar.gz
+      - run: tar --strip-components=1 -xzvf /opt/s2i.tar.gz -C /opt/
+      - run: mv /opt/s2i /usr/bin/
+      - env:
+          REDHAT_REGISTRY_LOGIN: ${{ secrets.REDHAT_REGISTRY_LOGIN }}
+        run: ${REDHAT_REGISTRY_LOGIN}
+      - run: s2i build --context-dir=${CONTEXT_DIR} --ref ${GITHUB_REF##*/} ${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY} ${NGINX_S2I_IMAGE} ${NGINX_IMAGE_NAME}:latest
```

Comment:
In `${GITHUB_REF##*/}`, what does the `##*/` mean or do or translate to? I'm not familiar with this style of syntax.

Reply:
I'm doing some weird parameter expansion here, effectively I just want everything after the last slash, as that value is normally fed by github as `ref/heads/potatoes` or something along those lines.

Here is some more info: https://stackoverflow.com/questions/19482123/extract-part-of-a-string-using-bash-cut-split/19482947#19482947